### PR TITLE
Fixing timezones for reports and order-editing

### PR DESCRIPTION
**Pull request by mintplugins**
#7016

### CODE REVIEW

File: assets/js/admin/reports/charts/index.js [hunk 1]
```diff
@@ -23,6 +23,7 @@ window.edd.renderChart = ( config ) => {
 	if ( isPieChart( config ) ) {
 		pieChartRender( config );
 	} else {
+		console.log( config );
```
Comment (JJJ) [inline assets/js/admin/reports/charts/index.js]: 😁

**Comment by mintplugins** [comment]
Note that these merge conflicts are all just built file issues since I am currently running in `npm run dev` mode.

File: includes/date-functions.php [hunk 2]
```diff
@@ -41,6 +41,70 @@ function edd_get_timezone_id() {
...
+
+	// Convert the timezone (and thus, also the time) from the WP/EDD Timezone to the UTC equivalent.
+	$utc_timezone = new DateTimeZone( 'utc' );
+	$edd_date_object->setTimezone( $utc_timezone );
+
+	return $edd_date_object;
+}
+
+/**
+ * Accept an EDD date object set in UTC, and get the WP/EDD Timezoneequivalent version of it.
```
Comment (nosegraze) [inline includes/date-functions.php]: Missing a space between `Timezone` and `equivalent`

File: includes/date-functions.php [hunk 3]
```diff
@@ -41,6 +41,70 @@ function edd_get_timezone_id() {
...
+	if ( ! $edd_date_object instanceof $instance_check ) {
+		return false;
+	}
+
+	// If you passed a date object to this function that isn't set to UTC, that is incorrect usage.
+	if ( 'UTC' !== $edd_date_object->format( 'T' ) ) {
+		return false;
+	}
+
+	// Convert the timezone (and this, also the time) from UTC to the WP/EDD Timezone.
```
Comment (nosegraze) [inline includes/date-functions.php]: Should `and this` be `and thus` ?

File: includes/reports/data/charts/v2/class-dataset.php [hunk 4]
```diff
@@ -266,6 +266,8 @@ public function parse_data_for_output( $data ) {
 						'y' => $values[1],
 					);
 				} else {
+					print_r( $values );
```
Comment (nosegraze) [inline includes/reports/data/charts/v2/class-dataset.php]: Is this still meant to be here?

File: includes/reports/reports-functions.php [hunk 5]
```diff
@@ -506,7 +510,6 @@ function get_dates_filter_options() {
  * }
  */
 function get_dates_filter( $values = 'strings', $timezone = null ) {
-	$date  = EDD()->utils->date( 'now', $timezone, false );
 	$dates = parse_dates_for_range( $date );
```
Comment (nosegraze) [inline includes/reports/reports-functions.php]: The removal of `$date` above is throwing an undefined variable notice here.

**Comment by nosegraze** [comment]
Left a couple comments about unimportant things. Then I also pointed out one actual problem. I haven't actually finished testing yet -- I'll continue once that's addressed.

**Comment by mintplugins** [comment]
@nosegraze Thanks! Those should all be fixed up now.

**Comment by nosegraze** [comment]
Getting this error now:

```
[04-Nov-2019 14:09:51 UTC] PHP Warning:  array_key_exists(): The first argument should be either a string or an integer in C:\wamp\www\edd3\wp-content\plugins\easy-digital-downloads\includes\reports\reports-functions.php on line 557
```

I believe it's just one missed instance of changing the `parse_dates_for_range()` arguments in `class-stats.php`, the `set_date_ranges()` method:

```php
$this->date_ranges[ $range ] = Reports\parse_dates_for_range( $date, $range );
```

**Comment by mintplugins** [comment]
@nosegraze Which screen do you get that error on? I just want to replicate it here if I can.

**Pull request merged**